force-app/main/default/classes/RD2_ERecurringDonationsSelector.cls

```diff
@@ -75,7 +75,8 @@ public inherited sharing class RD2_ERecurringDonationsSelector {
                                 + 'npe03__Date_Established__c, '
                                 + 'npe03__Next_Payment_Date__c, '
                                 + 'npe03__Total_Paid_Installments__c, '
-                                + 'Lastmodifieddate '
+                                + 'Lastmodifieddate, '
+                                + '(SELECT Active__c, Campaign__c, Campaign__r.Name, StartDate__c, EndDate__c, DayOfMonth__c, InstallmentAmount__c, InstallmentFrequency__c, InstallmentPeriod__c, PaymentMethod__c, RecurringDonation__c, IsPause__c, StatusReason__c FROM RecurringDonationSchedules__r) '
```

Comment:
Note that `RecurringDonationSchedule__c` is a protected custom object. An end user will not be able to query that object in code that is using `with sharing`. This particular class uses `inherited sharing` (as it should), but it's possible this method gets called from a controller that is using `with sharing`. I'm pretty sure that will cause the query to fail because the customer cannot see that object. 

You may need to make a separate query against the schedule object using an inner class (within this class) where the inner class explicitly uses 'without sharing'. You'd have to refactor to use two queries rather than a single query with a sub-query.

I do not recommend modifying this class or the controller class to use `without sharing`.

Reply:
@force2b Based on [[Technical Design Review] W-11379845 - NPSP](https://salesforce.quip.com/x8KIAjUPoJ8C) > Testing Considerations > Create Environment to test: The Admin will have to allow READ access to some objects to the partner profile, including Recurring Donation Schedules.
RD2_ETableController is public with sharing.
It's not failing the query and we are getting the information needed to process and get the final donation date.
It's working with 'with sharing'.